From 63dbd9d08de2713dff5146e1ba075a0ae5cdfb12 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:02:36 -0400
Subject: [usb] xhci: kill a conditional when toggling cycle

Message-id: <1410887463-20674-85-git-send-email-dzickus@redhat.com>
Patchwork-id: 95149
O-Subject: [RHEL7 PATCH 084/591] usb: xhci: kill a conditional when toggling cycle
Bugzilla: 1032371
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1032371

commit 4e341818ee9ee8d435977131cc7ce9e89c053fa6
Author: Dan Williams <dan.j.williams@intel.com>
Date:   Mon Oct 7 11:58:34 2013 -0700

    usb: xhci: kill a conditional when toggling cycle

    Perform an unconditional toggle of the cycle bit with 'xor'.

    Signed-off-by: Dan Williams <dan.j.williams@intel.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci-ring.c b/drivers/usb/host/xhci-ring.c
index afcaa4e..4f2a1ff 100644
--- a/drivers/usb/host/xhci-ring.c
+++ b/drivers/usb/host/xhci-ring.c
@@ -178,7 +178,7 @@ static void inc_deq(struct xhci_hcd *xhci, struct xhci_ring *ring)
    if (ring->type == TYPE_EVENT &&
      last_trb_on_last_seg(xhci, ring,
       ring->deq_seg, ring->dequeue)) {
-    ring->cycle_state = (ring->cycle_state ? 0 : 1);
+    ring->cycle_state ^= 1;
    }
    ring->deq_seg = ring->deq_seg->next;
    ring->dequeue = ring->deq_seg->trbs;
-- 
1.7.1